src/options/options.js

```diff
@@ -18,63 +27,92 @@ document.addEventListener("alpine:init", () => {
+            headers: headers,
+          });
           const { data, included } = await response.json();
+
+          // Give users with API token chance to switch to OAuth
+          if (!!accessTokenFromStorage) this.showLogin = false;
+
           workspaces = data;
           repositories = included.filter((item) => item.type === "repository");
         } catch (err) {
           console.error(err);
         }
       }
-      this.tokenCheckStatus.success = true;
-      this.token = tokenFromStorage;
+      this.authenticationCheckStatus.success = true;
+      this.token = apiKeyFromStorage;
+      this.accessToken = accessTokenFromStorage;
       this.selectedWorkspaceSlug = selectedWorkspaceSlugFromStorage;
       this.workspaces = workspaces;
       this.repositories = repositories;
     },
     async fetchWorkspaces() {
+      const url = new URL(`${ORBIT_API_ROOT_URL}/workspaces`);
+
+      const { params, headers } = configureRequest({
+        ACCESS_TOKEN: this.accessToken,
+        API_TOKEN: this.token,
+      });
+
+      url.search = params.toString();
+
       try {
-        const response = await fetch(
-          `${ORBIT_API_ROOT_URL}/workspaces?api_key=${this.token}`,
-          {
-            headers: { ...ORBIT_HEADERS },
-          }
-        );
+        const response = await fetch(url, {
+          headers: headers,
+        });
         if (!response.ok) {
-          this.tokenCheckStatus.success = false;
-          this.tokenCheckStatus.message = "The token is invalid.";
+          this.showLogin = true;
+          this.authenticationCheckStatus.success = false;
+          this.authenticationCheckStatus.message =
+            "Failed to authenticate, please try again.";
           return;
         }
         const { data, included } = await response.json();
         this.workspaces = data;
         this.repositories = included.filter(
           (item) => item.type === "repository"
         );
-        this.tokenCheckStatus.success = true;
-        this.tokenCheckStatus.message =
-          "The token is valid, please select a workspace.";
+        this.authenticationCheckStatus.success = true;
+        this.authenticationCheckStatus.message =
+          "Signed in successfully. Please select a workspace.";
+
+        if (!!this.accessToken) {
+          this.showLogin = false;
+        } else {
+          this.authenticationCheckStatus.message = `WARNING: You are using an API token for authentication, which is deprecated. Use the button above to switch to the new sign in process.`;
+        }
       } catch (err) {
         console.error(err);
-        this.tokenCheckStatus.success = false;
-        this.tokenCheckStatus.message =
-          "There was an unexpected error while validating the token.";
+        this.showLogin = true;
+        this.authenticationCheckStatus.success = false;
+        this.authenticationCheckStatus.message =
+          "There was an unexpected error whilst signing in.";
```

Comment:
🇬🇧

Reply:
😱 Okay I'm in too deep. I don't actually know where the british-ism is. Don't tell me I'm supposed to be spelling "error" as "errour"

Reply:
hahahaha it’s _while_ vs. _whilst_ :D I was wondering when one should use one or the other, and it turns out (apparently) to be a British vs. American English thing: https://www.grammarly.com/blog/while-vs-whilst/

Reply:
Awesome awesome, thanks for the clarification ^^ Will fix (and apologise to my king & country 😔 ✊🏻 )

Reply:
Please please please leave it like this! I love the fact that codebases sometimes reflect who wrote this or that piece of code (through language, or dummy data, etc…). It makes for a much more enjoyable read!

Reply:
:O Oh, wholesome!!! Will do 😄

Reply:
(I realize the original message wasn’t super clear: it was not intended as “flagging this as a british-ism that should be changed”, but more “here’s a new one I didn’t know about!”)